### PR TITLE
Cohen Neural Network

### PR DESCRIPTION
**Pull request by MC-Student**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,11 +10,11 @@ repositories {
 }
 
 dependencies {
-    implementation files('libs/basic_neural_network-v0.5.1-jar-with-dependencies.jar')
     implementation 'org.jetbrains:annotations:24.0.1'
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.9.2'
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.9.2'
     testImplementation 'org.mockito:mockito-core:5.5.0'
+    implementation files('libs/basic_neural_network-v0.5.1-jar-with-dependencies.jar')
```
Comment (andrewoid) [inline build.gradle:17]: Why did you change the location?

File: src/main/java/smartsnake/neuralnetwork/cohen/LearnAndSave.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import java.io.IOException;
+
+public class LearnAndSave
+{
+    public static void main(String[] args) throws IOException
+    {
+        NeuralNetwork neuralNetwork = new NeuralNetwork(5, 256, 4);
+        NeuralNetworkDataFactory dataFactory = new NeuralNetworkDataFactory();
+
+        for (int i = 0; i < 10000; i++)
```
Comment (andrewoid) [inline src/main/java/smartsnake/neuralnetwork/cohen/LearnAndSave.java:19]: `50_000`

**Review by andrewoid — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (MC-Student) [reply build.gradle:17]: Because for some reason it wasn't recognizing the neural network library (it couldn't find NeuralNetwork when I tried to import, etc). This allowed me to reload the build.gradle (I tried rebuilding it but it didn't help so I did this) - 
No actual difference
